ls1intum/Artemis
# [Bugfix] File browser compress tree

### Checklist

### Motivation and Context



The file browser did not correctly show the compressed file tree for some folder combinations.

### Description


Rewrite the tree compression method to fix the bug.

### Steps for Testing


1. Log in to Artemis
2. Navigate to the Code Editor
3. Add, Remove and Rename Files and Folders at different positions in the file tree
4. Check that the compressed tree is displayed correctly

### Test Coverage




Nothing changed

### Screenshots



Without tree compression
<img width="528" alt="Bildschirmfoto 2020-12-03 um 20 22 28" src="https://user-images.githubusercontent.com/13920539/101077588-552fbd00-35a5-11eb-989f-d90887aa0b7a.png">
With tree compression
<img width="529" alt="Bildschirmfoto 2020-12-03 um 20 22 34" src="https://user-images.githubusercontent.com/13920539/101077594-5660ea00-35a5-11eb-99e1-68a91e92a57c.png">
